src/components/Counter.js

```diff
@@ -3,37 +3,47 @@ import { connect } from 'react-redux';
 import { increment, decrement } from '../actions';
 
 class Counter extends Component {
-    incrementIfOdd = () => {
-        // Stretch Problem: Implement an increment function that
-        // only increments if the counter value is odd
+    incrementIfOdd = (props) => {
+        if  (this.props.count % 2 !== 0 ) {
+            return this.props.decrement(this.props.count + 1)
+        }
     };
 
     incrementAsync = () => {
         // Stretch Problem: Implement an increment function that
         // increments after waiting for one second
+       
+        const addOne = () => {
+            this.props.increment(this.props.count + 1)
+        }
+        setTimeout(
+            addOne, 
+            1000);
     };
 
+  
+
     render() {
         // Fill in the two button onClick methods
         // Upon clicking these buttons, the count
         // should decrement or increment accordingly
         return (
             <p>
                 Clicked: {this.props.count} times
-                <button onClick={() => {/* Fill me in */ }}>
+                <button onClick={() => {this.props.increment(this.props.count + 1)}}>
                     +
                 </button>
-                <button onClick={() => {/* Fill me in */ }}>
+                <button onClick={() => {this.props.decrement(this.props.count - 1)}}>
                     -
                 </button>
```

Comment:
While this `decrement` implementation does work well, the way you coded it is actually closer to a value assignment function than a true decrement by one function. As a result, there's no distinction between your `INCREMENT`` and `DECREMENT` reducer cases.

If you want to use this function purely for decrementing `count` by `1`, I would suggest the following:
• Do not pass in arguments for your function calls in your `Counter` component
• Same thing with your action creators file - don't take in parameters, and you don't even need to pass in an action payload (just pass in the action type)
• You can then update state in your reducer with a new object, like so:
```
return {
  ...state
  count: state.count - 1
}
```